### ISSUE TITLE
False positive InvalidArrayAccess in array index with array_rand 

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#266]
Expected: no error
Observed:
```
ERROR: InvalidArrayAccess - src/rand.php:6:17 - Cannot access value on array variable $vars using int offset - expecting 'x' or 'y'
    return [$c, $vars[$c]];  
```

```php
<?php

function test() {
    $vars = ['x' => 'a', 'y' => 'b'];
    $c = array_rand($vars);
    return [$c, $vars[$c]];
}
```

**Comment by muglug** [comment]
Thanks! Two bugs caught with this.

**Comment by TysonAndre** [comment]
After that commit, psalm complains if 2 arguments are provided to array_rand. The second is the number of requested elements.

```
<?php

function test() {
    $vars = ['x' => 'a', 'y' => 'b'];
    $c = array_rand($vars);
    return [$c, $vars[$c]];
}

/** @return string|string[] */
function x(int $n) {
    $vars = ['x' => 'a', 'y' => 'b'];
    return array_rand($vars, $n);  // incorrectly emitted: TooManyArguments: Too many arguments for method array_rand 
}
```

**Comment by muglug** [comment]
Those signature discrepancies have been fixed in https://github.com/vimeo/psalm/commit/a7b32beb4edfbdf268334678bfbfaaad5662c49b

And I maintain that this is one of the worst signatures in heavy usage (though `get_class(null)` gets a special shoutout)

**Comment by TysonAndre** [comment]
Thanks!
